tensorboard/plugins/debugger_v2/debug_data_provider.py

```diff
@@ -43,41 +43,51 @@
 STACK_FRAMES_BLOB_TAG_PREFIX = "stack_frames"
 
 
-def alerts_run_tag_filter(run, begin, end):
+def alerts_run_tag_filter(run, begin, end, alert_type=None):
     """Create a RunTagFilter for Alerts.
 
     Args:
       run: tfdbg2 run name.
       begin: Beginning index of alerts.
       end: Ending index of alerts.
+      alert_type: Optional alert type, used to restrict retrieval of alerts
+        data to a single type of alerts.
 
     Returns:
       `RunTagFilter` for the run and range of Alerts.
     """
-    # TODO(cais): Add filter for alert type.
-    return provider.RunTagFilter(
-        runs=[run], tags=["%s_%d_%d" % (ALERTS_BLOB_TAG_PREFIX, begin, end)],
-    )
+    tag = "%s_%d_%d" % (ALERTS_BLOB_TAG_PREFIX, begin, end)
+    if alert_type is not None:
+        tag += "_%s" % alert_type
+    return provider.RunTagFilter(runs=[run], tags=[tag])
```

Comment:
OK, but recall that if this really is intended to be substitutable with
a generic data provider then each possible value of `alert_type` will
need to have a separate blob stored, duplicating the data. If this is
fine with you, then proceed.

Reply:
The data duplication is fine with me. When we work on the BLOB storage, we can find ways to minimize the amount of data duplication by only allowing querying detailed alerts data of a given type at a time.